code/game/gamemodes/cult/cult_structures.dm

```diff
@@ -2,29 +2,135 @@
+		return
+	var/choice = alert(user,"You study the schematics etched into the forge...",,"Shielded Robe","Flagellant's Rags","Nar-Sian Hardsuit")
+	switch(choice)
+		if("Shielded Robe")
+			var/obj/item/clothing/suit/cultrobes/cult_shield/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You work the forge as dark knowledge guides your hands, creating [N]!</span>"
+		if("Flagellant's Rags")
+			var/obj/item/clothing/suit/cultrobes/berserker/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You work the forge as dark knowledge guides your hands, creating [N]!</span>"
+		if("Nar-Sian Hardsuit")
+			new /obj/item/clothing/head/helmet/space/cult(get_turf(src))
+			var /obj/item/clothing/suit/space/cult/N = new(get_turf(src))
+			user << "<span class='cultitalic'>You work the forge as dark knowledge guides your hands, creating [N]!</span>"
+	cooldowntime = world.time + 2400
+	spawn(cooldowntime)
+	cooldowntime = 0
 
 /obj/structure/cult/pylon
 	name = "pylon"
-	desc = "A floating crystal that hums with an unearthly energy"
+	desc = "A floating crystal that slowly heals those faithful to Nar'Sie"
 	icon_state = "pylon"
 	luminosity = 5
+	var/heal_delay = 50
+	var/last_shot = 0
+	var/list/corruption = list()
+
+/obj/structure/cult/pylon/New()
+	SSobj.processing |= src
+	corruption += get_turf(src)
+	for(var/i = 0, i < 4, i++)
+		for(var/turf/T in corruption)
+			corruption |= T.GetAtmosAdjacentTurfs()	
+	..()
+
+/obj/structure/cult/pylon/Destroy()
+	SSobj.processing.Remove(src)
+	return ..()
+	
+/obj/structure/cult/pylon/process()
+	if((last_shot + heal_delay) <= world.time)
+		last_shot = world.time
+		for(var/mob/living/L in range(1, src))
+			if(iscultist(L))
+				var/mob/living/carbon/human/H = L
+				if(istype(H))
+					L.adjustBruteLoss(-1, 0)
+					L.adjustFireLoss(-1, 0)
+					L.updatehealth()
+				if(istype(L, /mob/living/simple_animal/hostile/construct))
+					var/mob/living/simple_animal/M = L
+					if(M.health < M.maxHealth)
+						M.adjustHealth(-2)
+		if(corruption.len)
+			var/turf/T = pick_n_take(corruption)
+			corruption -= T
+			if (istype(T, /turf/open/floor/engine/cult) || istype(T, /turf/open/space) || istype(T, /turf/open/floor/plating/lava))
+				return
+			T.ChangeTurf(/turf/open/floor/engine/cult)
 
 /obj/structure/cult/tome
-	name = "desk"
+	name = "archives"
 	desc = "A desk covered in arcane manuscripts and tomes in unknown languages. Looking at the text makes your skin crawl"
 	icon_state = "tomealtar"
 	luminosity = 1
 
+/obj/structure/cult/tome/attack_hand(mob/living/user)
+	if(!iscultist(user))
+		user << "<span class='warning'>You don't even begin to understand what these words mean...</span>"
+		return
+	if(cooldowntime)
+		user << "<span class='cultitalic'>The magic here is weak, it will be ready to use again in [getETA()]. </span>"
+		return
+	var/obj/item/weapon/paper/talisman/supply/N = new(get_turf(src))
```

Comment:
It could, but tomes are dirt cheap to get. Every conversion gives a tome and you can always make more with a tome talisman. Supply talismans are a super roundabout way of getting one but they're primarily for quickly getting a stun/EMP talisman or grabbing another shell/shard if you've run out.